### DISCUSSION TITLE
Building development version failed (maven)

### DISCUSSION THREAD

**Discussion by gerlingsm** [kitodo/kitodo-production#5525] [Q&A] [answered]
Hello everybody,

i was trying to build the latest kitodo version from the master branch according to the following instructions: [Build development version of Kitodo 3.x](https://kitodoproduction-dev.readthedocs.io/en/latest/developer/gettingstarted/development-version/).
But unfortunately  the generation of the SQL dump failed. 
More precisely, the command `mvn flyway:baseline -Pflyway` leads to the following error:
(I don't use mysql-community-server but mariadb, but I don't think the error is because of that.)

> [INFO] 
[INFO] -----------------< org.kitodo:kitodo-data-management >------------------
[INFO] Building Kitodo - Data Management 3.5.0-beta-1
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from central: https://repo1.maven.org/maven2/org/kitodo/kitodo-api/3.5.0-beta-1/kitodo-api-3.5.0-beta-1.pom
[WARNING] The POM for org.kitodo:kitodo-api:jar:3.5.0-beta-1 is missing, no dependency information available
Downloading from central: https://repo1.maven.org/maven2/org/kitodo/kitodo-api/3.5.0-beta-1/kitodo-api-3.5.0-beta-1.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.866 s
[INFO] Finished at: 2023-02-02T15:01:03+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project kitodo-data-management: Could not resolve dependencies for project org.kitodo:kitodo-data-management:jar:3.5.0-beta-1: Could not find artifact org.kitodo:kitodo-api:jar:3.5.0-beta-1 in central (https://repo1.maven.org/maven2) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException

Have I done something wrong here or forgotten something?
I am using Java11, on Ubuntu 22.04, with maven 3.6.3 and mariadb 10.6.11

Best regards

**Comment by markusweigelt** [comment, marked as answer]
Seems that the documentation page is not up to date. To run the `mvn flyway:baseline -Pflyway`, you need to build the dependencies of Kitodo.Production in your Maven repository.

`mvn clean install -DskipTests`

Then navigate to the Kitodo-DataManagement directory and call the `flyway` commands:

`cd ./Kitodo-DataManagement && mvn flyway:baseline -q  -Pflyway && mvn flyway:migrate -q -Pflyway`

In our [workflow](https://github.com/kitodo/kitodo-production/blob/master/.github/workflows/main.yml) you will also find a corresponding example.

Or there is a [script](https://github.com/slub/kitodo-production-docker/blob/main/build-git.sh) to build the Docker image resources of Kitodo.Production. 

However, you can also use the Docker image directly. https://github.com/slub/kitodo-production-docker
